esphome/components/animation/__init__.py

```
@@ -71,8 +71,11 @@ async def to_code(config):
         pos = 0
         for frameIndex in range(frames):
             image.seek(frameIndex)
+            if CONF_RESIZE in config:
+                image.thumbnail(config[CONF_RESIZE])
             frame = image.convert("RGB")
             pixels = list(frame.getdata())
+            assert len(pixels) == height * width
```

Can we raise an error instead? Assertion failures aren't very friendly to users...

---

Ok. done.